pipelines/rancher/ui-build.groovy

```diff
@@ -12,39 +15,48 @@ properties([
         jobsParameters.clusterName(),
         jobsParameters.projectName(),
         jobsParameters.tenantId(),
-        jobsParameters.agents(),
         string(name: 'custom_hash', defaultValue: '', description: 'Commit hash for bundle build from specific commit'),
         string(name: 'custom_url', defaultValue: '', description: 'Custom url for okapi'),
-        string(name: 'custom_tag', defaultValue: '', description: 'Custom tag for UI image')
+        string(name: 'custom_tag', defaultValue: '', description: 'Custom tag for UI bundle image')
     ])
 ])
 
+OkapiTenant tenant = new OkapiTenant(id: params.tenant_id)
 
-String okapi_domain = common.generateDomain(params.rancher_cluster_name, params.rancher_project_name, 'okapi', Constants.CI_ROOT_DOMAIN)
-String okapi_url = params.custom_url.isEmpty() ? "https://" + okapi_domain : params.custom_url
-String hash = params.custom_hash.isEmpty() ? common.getLastCommitHash(params.folio_repository, params.folio_branch) : params.custom_hash
-String tag = params.custom_tag.isEmpty() ? "${params.rancher_cluster_name}-${params.rancher_project_name}-${params.tenant_id}-${hash.take(7)}" : params.custom_tag
-String image_name = "${Constants.DOCKER_DEV_REPOSITORY}/platform-complete:${tag}" //TODO rename to folio-ui
+Project project_model = new Project(
+    clusterName: params.rancher_cluster_name,
+    projectName: params.rancher_project_name,
+    domains: [ui   : common.generateDomain(params.rancher_cluster_name, params.rancher_project_name, tenant.getId(), Constants.CI_ROOT_DOMAIN),
+              okapi: common.generateDomain(params.rancher_cluster_name, params.rancher_project_name, 'okapi', Constants.CI_ROOT_DOMAIN),
+              edge : common.generateDomain(params.rancher_cluster_name, params.rancher_project_name, 'edge', Constants.CI_ROOT_DOMAIN)]
+)
+
+Module ui_bundle = new Module(
+    name: "platform-complete", //TODO rename to ui-bundle
+    hash: params.custom_hash?.trim() ? params.custom_hash : common.getLastCommitHash(params.folio_repository, params.folio_branch)
+)
+
+ui_bundle.tag = params.custom_tag?.trim() ? params.custom_tag : "${project_model.getClusterName()}-${project_model.getProjectName()}-${tenant.getId()}-${ui_bundle.getHash().take(7)}"
+ui_bundle.imageName = "${Constants.DOCKER_DEV_REPOSITORY}/${ui_bundle.getName()}:${ui_bundle.getTag()}"
+
+String okapi_url = params.custom_url?.trim() ? params.custom_url : "https://" + project_model.getDomains().okapi
 
 ansiColor('xterm') {
-    if (params.refreshParameters) {
-        currentBuild.result = 'ABORTED'
-        error('DRY RUN BUILD, NO STAGE IS ACTIVE!')
-    }
-    node(params.agent) {
+    common.refreshBuidParameters(params.refresh_parameters)
+    node("jenkins-agent-java11") {
```

Comment:
Also possible change to jobsParameters.agents() ?

Reply:
Just to reduce parameters number for end users. Cause here it is not required to specify java version